tests/actions/test.transaction.js

```diff
@@ -99,4 +106,131 @@ describe('transactionActions', function() {
 
   });
 
+  describe('processPayment', function() {
+
+    function processPayment({fetch, productId,
+                             client=helpers.FakeBraintreeClient,
+                             payMethod=fakeCreditCard}) {
+      var deferredAction = transactionActions.processPayment({
+        productId: productId,
+        payMethod: payMethod,
+        braintreeToken: 'braintree-token',
+        fetch: fetch,
+        BraintreeClient: client,
+      });
+
+      helpers.doApiAction(deferredAction, dispatchSpy);
+    }
+
+    it('should create a subscription for recurring products', function() {
+      var fetch = helpers.fakeFetch();
+      processPayment({fetch: fetch, productId: 'mozilla-concrete-brick'});
+      assert.equal(fetch.firstCall.args[0].url, '/braintree/subscriptions/');
+    });
+
+    it('should create a sale for one-off products', function() {
+      var fetch = helpers.fakeFetch();
+      processPayment({fetch: fetch, productId: 'mozilla-foundation-donation'});
```

Comment:
For a later bug, but i'm not keen on test in payments-ui being dependent upon data inside payments-config. Ideally we can define a new set of values for the config for these tests.